src/main/java/org/wso2/carbon/connector/operations/EmailDelete.java

```diff
@@ -0,0 +1,79 @@
+import org.wso2.carbon.connector.connection.EmailConnectionManager;
+import org.wso2.carbon.connector.connection.EmailConnectionPool;
+import org.wso2.carbon.connector.connection.MailBoxConnection;
+import org.wso2.carbon.connector.core.AbstractConnector;
+import org.wso2.carbon.connector.exception.ContentBuilderException;
+import org.wso2.carbon.connector.exception.EmailConnectionException;
+import org.wso2.carbon.connector.exception.EmailConnectionPoolException;
+import org.wso2.carbon.connector.exception.EmailNotFoundException;
+import org.wso2.carbon.connector.exception.InvalidConfigurationException;
+import org.wso2.carbon.connector.utils.ConfigurationUtils;
+import org.wso2.carbon.connector.utils.EmailConstants;
+import org.wso2.carbon.connector.utils.EmailOperationUtils;
+import org.wso2.carbon.connector.utils.Error;
+import org.wso2.carbon.connector.utils.ResponseHandler;
+
+import javax.mail.Flags;
+
+import static java.lang.String.format;
+
+/**
+ * Deletes an Email
+ */
+public class EmailDelete extends AbstractConnector {
+
+    @Override
+    public void connect(MessageContext messageContext) {
+
+        String errorString = "Error occurred while deleting email from folder: %s. %s";
+
+        String folder = (String) getParameter(messageContext, EmailConstants.FOLDER);
+        String emailID = (String) getParameter(messageContext, EmailConstants.EMAIL_ID);
+        EmailConnectionPool pool = null;
+        MailBoxConnection connection = null;
+        try {
+            String connectionName = ConfigurationUtils.getConnectionName(messageContext);
+            pool = EmailConnectionManager.getEmailConnectionManager().getConnectionPool(connectionName);
+            connection = (MailBoxConnection) pool.borrowObject();
+            boolean status = EmailOperationUtils.changeEmailState(connection, folder, emailID, Flags.Flag.DELETED,
+                    true);
+            ResponseHandler.generateOutput(messageContext, status);
+        } catch (EmailConnectionException | EmailConnectionPoolException e) {
+            ResponseHandler.setErrorsInMessage(messageContext, Error.CONNECTIVITY);
+            handleException(format(errorString, folder, e.getMessage()), e, messageContext);
+        } catch (EmailNotFoundException e) {
+            ResponseHandler.setErrorsInMessage(messageContext, Error.EMAIL_NOT_FOUND);
+            handleException(format(errorString, folder, e.getMessage()), e, messageContext);
+        } catch (InvalidConfigurationException e) {
+            ResponseHandler.setErrorsInMessage(messageContext, Error.INVALID_CONFIGURATION);
+            handleException(format(errorString, folder, e.getMessage()), e, messageContext);
+        } catch (ContentBuilderException e) {
+            ResponseHandler.setErrorsInMessage(messageContext, Error.RESPONSE_GENERATION);
+            handleException(format(errorString, folder, e.getMessage()), e, messageContext);
+        } finally {
+            if (pool != null) {
```

Comment:
Do we need this null check?

Reply:
pool may become null if an exception is thrown.